src/main/java/graphql/ExecutionResultImpl.java

```diff
@@ -41,7 +41,7 @@ private ExecutionResultImpl(boolean dataPresent, Object data, List<? extends Gra
         if (errors != null && !errors.isEmpty()) {
             this.errors = Collections.unmodifiableList(new ArrayList<>(errors));
         } else {
-            this.errors = Collections.emptyList();
+            this.errors = null;
```

Comment:
We chose to have the `toSpecification` to allow you to build a result that perfectly follows the spec